changes/700.feature.md

```diff
@@ -0,0 +1 @@
+feat: Add class KernelLifecycleEventReason to manage KernelEvent.reason
```

Comment:
This is not a valid style for changelogs. Please refer [changes/README](https://github.com/lablup/backend.ai/blob/main/changes/README.md).